spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/BindableProxyFactory.java

```diff
@@ -218,9 +230,10 @@ public void bindInputs(BindingService bindingService) {
 				if (log.isDebugEnabled()) {
 					log.debug(String.format("Binding %s:%s:%s", this.namespace, this.type, inputTargetName));
 				}
-				bindingService.bindConsumer(boundTargetHolder.getBoundTarget(), inputTargetName);
+				bindings.addAll(bindingService.bindConsumer(boundTargetHolder.getBoundTarget(), inputTargetName));
 			}
 		}
+		return bindings;
```

Comment:
Shound't we worry here about an *immutability* of the collection to return?

Reply:
may be, I'll look how it is used or can be used